src/components/buttons/Button.jsx

```diff
@@ -0,0 +1,46 @@
+import PropTypes from 'prop-types';
+import MUIButton from '@mui/material/Button';
+import { StyledEngineProvider } from '@mui/material';
+import css from './button.module.scss';
+
+const Button = ({ type, disabled, children }) => {
+    let variant;
+    let cssClass;
+    switch (type) {
```

Comment:
lets refactor this with 'classnames' lib

with it we can apply styles according to prop 

```
const classes = cx(
      styles.button,
      styles[size],
      styles[`${variant}_${mode}`],
      {
        [styles.custom_style]: isTrue,
      },
      className,
    );
```


and then apply classes like so 
```
<MUIButton
                disableElevation
                disabled={disabled}
                disableRipple
                variant={variant}
                className={classes}
            >
```